src/main/java/com/epam/aidial/core/service/ResourceService.java

```diff
@@ -229,7 +229,9 @@ public void computeResource(ResourceDescription descriptor, Function<String, Str
         try (var ignore = lockService.lock(redisKey)) {
             String body = getResource(descriptor, false);
             String updatedBody = fn.apply(body);
-            putResource(descriptor, updatedBody, true, false);
+            if (updatedBody != null) {
+                putResource(descriptor, updatedBody, true, false);
```

Comment:
yes, makes sense.

Reply:
Otherwise it is expected to fail

Reply:
in most current cases null means - resource absent, we don't need to delete it. Let's not rush in this direction right away.
Added resource body check to avoid redundant updates